app/models/attached_to_embeddable.rb

```diff
@@ -15,40 +15,88 @@ def self.included(clazz)
   NO_EMBEDDABLE_VALUE = "no-embeddable"
   NO_EMBEDDABLE_SELECT = [NO_EMBEDDABLE_LABEL, NO_EMBEDDABLE_VALUE]
 
+  # Model can be attached to the next embeddable present on a given page.
+  # It will be calculated dynamically when needed (see embeddable method below).
+  # That way, an author can use embeddable item positioning to wrap other embeddables / attach them to each other.
+  NEXT_EMBEDDABLE_LABEL = I18n.t('NEXT_EMBEDDABLE')
+  NEXT_EMBEDDABLE_VALUE = "next-embeddable"
+  NEXT_EMBEDDABLE_SELECT = [NEXT_EMBEDDABLE_LABEL, NEXT_EMBEDDABLE_VALUE]
+
+  def embeddable
+    if attached_to_next_interactive
+      embeddables = page.embeddables
```

Comment:
Can you think of a way to do this without loading the actual embeddable objects?  I'm worried that on long pages this will be slow if it happens a lot. If embeddables is using 'acts_as_list' then it might have a next method that efficiently finds the next one.  But I'd guess the embeddables list is not a simple acts_as_list association.

Reply:
That's a good point. I should be able to come up with something more efficient.

Reply:
@scytacki, fixed here:
https://github.com/concord-consortium/lara/pull/353/commits/6d460934faae3d026881ea24c24eb7f26a5ab7af
Example SQL query:
```
PageItem Load (0.8ms)  SELECT `page_items`.* FROM `page_items` WHERE `page_items`.`embeddable_id` = 26 AND `page_items`.`embeddable_type` = 'Embeddable::EmbeddablePlugin' LIMIT 1
  PageItem Load (0.6ms)  SELECT `page_items`.* FROM `page_items` WHERE (`page_items`.`interactive_page_id` = 538 AND position > 14) ORDER BY page_items.position ASC LIMIT 1
  Embeddable::Xhtml Load (0.7ms)  SELECT `embeddable_xhtmls`.* FROM `embeddable_xhtmls` WHERE `embeddable_xhtmls`.`id` = 16 LIMIT 1
```
So, it's loading page_item, then the next one, and finally the next embeddable. Seems reasonable.